sssd_test_framework/misc/__init__.py

```diff
@@ -17,16 +17,24 @@ def attrs_parse(lines: list[str], attrs: list[str] | None = None) -> dict[str, l
     :rtype: dict[str, list[str]]
     """
     out: dict[str, list[str]] = {}
-    for line in lines:
-        line = line.strip()
+    i = 0
+    while i < len(lines):
+        line = lines[i].strip()
```

Comment:
This way it is much better. But you are still loosing trailing spaces. Do not strip here.